src/utils/toFormData.js

```diff
@@ -13,7 +13,7 @@ function serialize(params, obj, traditional, scope) {
         key =
           scope +
           '[' +
-          (hash || type == 'object' || type == 'array' ? key : '') +
+          (hash || type === 'object' || type === 'array' ? key : '') +
```

Comment:
Only make changes that you PR refers to. Rever this file back to master. Also there is an existing PR fixing these issues.